utilities/fault_injection_fs.cc

```diff
@@ -165,15 +161,20 @@ IOStatus TestFSWritableFile::Append(const Slice& data, const IOOptions& options,
   if (!fs_->IsFilesystemActive()) {
     return fs_->GetError();
   }
+
+  IOStatus s = fs_->MaybeInjectThreadLocalWriteError(options);
+  if (!s.ok()) {
+    return s;
+  }
+
   if (target_->use_direct_io()) {
-    target_->Append(data, options, dbg).PermitUncheckedError();
+    s = target_->Append(data, options, dbg);
   } else {
     state_.buffer_.append(data.data(), data.size());
     state_.pos_ += data.size();
     fs_->WritableFileAppended(state_);
   }
-  IOStatus io_s = fs_->InjectWriteError(state_.filename_);
-  return io_s;
```

Comment:
If the actual write operation succeeds, we should not inject error IMO. Similar for below change.